src/AppBundle/Service/RoleManager.php

```diff
@@ -76,4 +78,33 @@ public function loggedInUserCanCreateUserWithRole(string $role): bool
                 $role !== Roles::ASSISTANT)
         ;
     }
+
+    public function loggedInUserCanChangeRoleOfUsersWithRole(User $user, string $role): bool
+    {
+        // Teamleaders can't change the role of admins
+        $loggedInAsAdmin = $this->authorizationChecker->isGranted(Roles::ADMIN);
+        $tryingToChangeAdmin = $this->userIsGranted($user, Roles::ADMIN);
+
+        return
+            ($loggedInAsAdmin || !$tryingToChangeAdmin) &&
+            $this->canChangeToRole($role);
+    }
+
+    public function userIsGranted(User $user, string $role): bool
+    {
+        if (empty($user->getRoles())) {
+            return false;
+        }
+
+        $roles = array(
+            'ROLE_USER',
+            'ROLE_ADMIN',
+            'ROLE_SUPER_ADMIN',
+            'ROLE_HIGHEST_ADMIN',
+        );
+
+        $userRole = $user->getRoles()[0]->getRole();
```

Comment:
Dette har jeg vel fiksa fra før? Jeg kan flytte if-statementen ned til $userRole for å øke lesbarheten da.